pkg/ui/src/util/locations.ts

```diff
@@ -89,3 +89,44 @@ export function findOrCalculateLocation(locations: LocationTree, locality: Local
   centroid = vector.mult(centroid, 1 / childLocations.length);
   return { longitude: centroid[0], latitude: centroid[1] };
 }
+
+/*
+ * getConfigStats finds how many nodes are missing locality or location config.
+ */
+export function getConfigStatus(
+  localityTree: LocalityTree, locationTree: LocationTree,
+): NodeMapConfigStatus {
+  let totalNodes = 0;
+  let withLocality = 0;
+  let withLocation = 0;
+  function recur(tree: LocalityTree) {
```

Comment:
it may be a moot point, but I bet this method would be much clearer if it just operated on the list of nodes, rather than the locality tree.

Reply:
Good call. This function used to do more, but now the recursion is completely unnecessary.